internal/server/storage.go

```diff
@@ -28,6 +29,12 @@ type Storage interface {
 	CreateWithdrawal(ctx context.Context, userID int64, order string, sum float64) (*models.Withdrawal, error)
 	GetWithdrawalsByUserID(ctx context.Context, userID int64) ([]models.Withdrawal, error)
 
+	// Transactional withdrawal - проверяет баланс и создает списание в одной транзакции
+	ProcessWithdrawal(ctx context.Context, userID int64, order string, sum float64) (*models.Withdrawal, error)
+
+	// Атомарное обновление статуса заказа и баланса пользователя
+	UpdateOrderStatusAndBalanceTx(ctx context.Context, orderNumber string, status string, accrual *float64, userID int64, newCurrent, withdrawn float64) error
```

Comment:
```suggestion
	UpdateOrderStatusAndBalance(ctx context.Context, orderNumber string, status string, accrual *float64, userID int64, newCurrent, withdrawn float64) error
```

потому что для вызывающего пакета неважно, что внутри функции выполняется транзакция